flycheck.el

```diff
@@ -9353,7 +9353,7 @@ See URL `http://pear.php.net/package/PHP_CodeSniffer/'."
             ;; Some files are not detected correctly
             ;; so it is necessary to pass the extension.
             (eval
-             (when (file-name-extension (buffer-file-name))
+             (when (buffer-file-name)
                (concat "--extensions=" (file-name-extension (buffer-file-name)))))
```

Comment:
You *also* need the extension, since the buffer might not have when; that's why I suggested `-when-let*`